environ/process/eigen_cluster/prepare_eigen_cluster.py

```diff
@@ -0,0 +1,325 @@
+"""
+Script to generate the eigenvector centrality for atomic swaps.
+"""
+
+import glob
+import os
+import warnings
+
+from typing import Literal
+import networkx as nx
+import numpy as np
+import pandas as pd
+from tqdm import tqdm
+
+from data.constants import NETWORK_DATA_PATH, BETWEENNESS_DATA_PATH
+
+# Ignore warnings
+warnings.filterwarnings("ignore")
+
+
+def _load_in_data_lst(
+    file_root: str,
+    filter_name: str = "",
+) -> list:
+    """
+    Function to generate a list of dataframes from a given path
+    """
+
+    # get the list of files
+    path = file_root
+    file_lst = glob.glob(path + "/*.csv")
+
+    # isolate the file with specific version
+    file_name_lst = (
+        [file_name for file_name in file_lst if filter_name == file_name.split("_")[-2]]
+        if filter_name
+        else file_lst
+    )
+
+    return file_name_lst
+
+
+def _generate_node_edge(
+    file_name: str,
+    edge_col: list[str],
+    weight_col: list[str],
+    exclude_special_route: bool = True,
+    dict2str: bool = False,
+    aggreate_weight: bool = False,
+    convert_undirected: bool = False,
+) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Function to generate the node and edge arrays
+    """
+
+    # load the data
+    df_network = pd.read_csv(file_name)
+
+    if exclude_special_route:
+        # Exclude "LOOP" AND "SPOON", and "Error" in intermediary
+        df_network = df_network[
+            (df_network["label"] == "0") & (df_network["intermediary"] != "Error")
+        ].copy()
+
+    if dict2str:
```

Comment:
this  dict2str shouldn't be part of this function, but should be part of preprocessing

the input data of the function should be more or less fixed.-  if not conformed, then preprocessing on a case-by-case basis. you shouldn't (and cannot) exhaust all pre-processing possibilities in the normal processing funciton